src/air/public_input.cairo

```diff
@@ -16,15 +20,101 @@ struct PublicInput {
     log_n_steps: felt252,
     rc_min: felt252,
     rc_max: felt252,
+    layout: felt252,
+    dynamic_params: Array<felt252>,
     segments: Array<SegmentInfo>,
     padding_addr: felt252,
     padding_value: felt252,
     main_page: Page,
-    continuous_page_headers: Span<ContinuousPageHeader>
+    continuous_page_headers: Array<ContinuousPageHeader>
 }
 
 #[generate_trait]
 impl PublicInputImpl of PublicInputTrait {
+    // Computes the hash of the public input, which is used as the initial seed for the Fiat-Shamir heuristic.
+    fn get_public_input_hash(self: @PublicInput) -> u256 {
+        // Main page hash.
+        let mut main_page_hash_state = PedersenTrait::new(0);
```

Comment:
Since this logic (pedersen hashing multiple field elements) is going to be used in other places in the code, I created a common function for it, and will refactor this in a future PR